packages/suite/src/views/settings/SettingsDebug/Passwords.tsx

```diff
@@ -0,0 +1,256 @@
+    `Unlock ${title} for user ${username}?`;
+
+const PasswordEntryComponent = ({
+    username,
+    title,
+    nonce,
+    password,
+    devicePath,
+}: PasswordEntryProps) => {
+    const [decodedPassword, setDecodedPassword] = useState('');
+    const [inProgress, setInProgress] = useState(false);
+
+    const decode = useCallback(() => {
+        if (inProgress) return;
+        setInProgress(true);
+        TrezorConnect.cipherKeyValue({
+            device: { path: devicePath },
+            override: true,
+            useEmptyPassphrase: true,
+            path: PATH,
+            key: getDisplayKey(title, username),
+            value: nonce,
+            encrypt: false,
+            askOnEncrypt: false,
+            askOnDecrypt: true,
+        })
+            .then(result => {
+                if (result.success) {
+                    const decrypted = metadataUtils.decrypt(
+                        Buffer.from(password.data),
+                        Buffer.from(result.payload.value, 'hex'),
+                    );
+                    setDecodedPassword(decrypted);
+                }
+            })
+            .finally(() => {
+                setInProgress(false);
+            });
+    }, [username, title, inProgress, devicePath, nonce, password]);
+
+    return (
+        <>
+            <PasswordEntryRow>
+                <PasswordEntryBody>
+                    <PasswordEntryTitle>{title}</PasswordEntryTitle>
+                    <PasswordEntryUsername>{username}</PasswordEntryUsername>
+                    <PasswordEntryPassword>
+                        {!decodedPassword ? (
+                            <Button onClick={decode} type="button" variant="tertiary">
+                                {inProgress ? '....' : 'decode'}
+                            </Button>
+                        ) : (
+                            <>{decodedPassword}</>
+                        )}
+                    </PasswordEntryPassword>
+                </PasswordEntryBody>
+            </PasswordEntryRow>
+        </>
+    );
+};
+
+export const PasswordManager = () => {
+    // todo: filename is saved only locally. for production grade state of this feature we will of course need to save it into app state.
+    const [fileName, setFileName] = useState('');
+    const [providerConnecting, setProviderConnecting] = useState(false);
+    const [fetchingPasswords, setFetchingPasswords] = useState(false);
+
+    const { device } = useSelector(state => ({
+        metadata: state.metadata,
+        device: state.device.selectedDevice,
+    }));
+
+    const selectedProvider = useSelector(selectSelectedProviderForPasswords);
+
+    const { fetchPasswords, connectProvider, disconnectProvider } = useActions({
+        connectProvider: metadataActions.connectProvider,
+        disconnectProvider: metadataActions.disconnectProvider,
+        fetchPasswords: metadataActions.fetchPasswords,
+    });
```

Comment:
Don't use `useActions` please, just standard `dispatch`.

Reply:
0a01194abc9bb8aa7d4067074c5e7be7ee1eaa05

Reply:
unnecessary dependency, see the validation error

Reply:
b2a794e7dafaffecb39d3f812d70c779b5fb2c6d